feat: landing & config page UX improvements

SgtPooki opened the pull request:
## Title

<!---
The title of the PR will be the commit message of the merge commit, so please make sure it is descriptive enough.
We utilize the Conventional Commits specification for our commit messages. See <https://www.conventionalcommits.org/en/v1.0.0/#specification> for more information.
The commit tag types can be of one of the following: feat, fix, deps, refactor, chore, docs. See <https://github.com/ipfs/helia/blob/main/.github/workflows/main.yml#L184-L192>
The title must also be fewer than 72 characters long or it will fail the Semantic PR check. See <https://github.com/ipfs/helia/blob/main/.github/workflows/semantic-pull-request.yml>
--->

feat: landing & config page UX improvements

## Description

<!--
Please write a summary of your changes and why you made them.
Please include any relevant issues in here, for example:
Related https://github.com/ipfs/helia/issues/ABCD.
Fixes https://github.com/ipfs/helia/issues/XYZ.
-->

Fixes https://github.com/ipfs-shipyard/service-worker-gateway/issues/58

### Summary of changes

1. Add explainer with links to helper-ui (landing page)
2. Allow setting dnsResolvers in config.
3. Shows default urls in config
4. Requires at least 1 dns resolver/router/gateway
5. Adds "reset" button so users can get back to defaults if they bork things (~~TODO: need to fix this so it doesn't require a refresh, which will not work on first-hit for subdomain~~ this todo is done.)

### Landing page view

<img width="1189" alt="image" src="https://github.com/ipfs-shipyard/service-worker-gateway/assets/1173416/5cd09358-305b-4c0a-ac74-016632a3a950">

### Config page view

<img width="800" alt="image" src="https://github.com/ipfs-shipyard/service-worker-gateway/assets/1173416/b73b06f4-cd22-49c1-b0f8-d36688074115">


## Notes & open questions

<!--
Any notes, remarks or open questions you have to make about the PR which don't need to go into the final commit message.
-->

FYI that inputs are still JSON based for now.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [ ] I have added tests that prove my fix is effective or that my feature works

SgtPooki committed:
feat: add explainer with links to helper-ui

SgtPooki committed:
feat: allow setting dnsResolvers in config. show defaults. require urls

SgtPooki committed:
fix: remove second param from root.render

SgtPooki committed:
fix: e2e tests and redirect page load-content btn

SgtPooki reviewed (commented):
self review.. i'm going to try to provide a config-provider so we can call reset/save/etc there and remove the localStorage items

SgtPooki committed:
fix: reset config values without page reload

SgtPooki committed:
chore: remove unnecessary code

2color commented on src/index.tsx:
```diff
@@ -31,6 +31,5 @@ root.render(
       <App />
     </RouterProvider>
-  </React.StrictMode>,
-  container
```
Is this intentional?

SgtPooki replied:
yep! the second param is not needed for react-dom, but was for preact.

SgtPooki committed:
Merge branch 'main' into 58-improve-ux-of-the-root-domain-landing-page

lidel committed:
refactor: colors and about section

- moved About to the bottom and made it more friendly
- adjusted colors to improve contrast
- header to match other web utiities we have

lidel reviewed (approved):
Thanks! Content lgtm, but it felt a bit distracting, so played a bit and ended up moving it to bottom.

Pushed a commit with some additional changes:

- moved About to the bottom and made it more friendly for someone who wants to learn how things work
- adjusted colors to improve contrast and indicate when button is enabled
- tweaked the header to include (beta) and match style of other web utiities we have like https://explore.ipld.io/ or https://cid.ipfs.tech/

@SgtPooki @2color  If these look good, feel free to merge or rephrase if anything sounds off.

### Preview

> ![Screen Shot 2024-05-14 at 22 13 49](https://github.com/ipfs/service-worker-gateway/assets/157609/ac8984a0-996f-401b-98ab-f294ec53c94f)

SgtPooki commented:
@lidel that looks a lot cleaner. i'll fix up the CI and get this merged.

SgtPooki committed:
Merge branch 'main' into 58-improve-ux-of-the-root-domain-landing-page

SgtPooki committed:
test: fix header text comparison
